### PR TITLE
hackathons/usoc23: Add basic application porting session

### PR DESCRIPTION
**Pull request by StefanJum**
Closes: #258

### CODE REVIEW

**Comment by StefanJum** [comment]
On Thu, Jul 13, 2023 at 08:25:46AM -0700, Eduard Vintilă wrote:
> @eduardvintila commented on this pull request.
> 
> Hey, @StefanJum! Just some small fixes for the moment :) :
> 
> In the `Example of External Library` section:
> 1. Modify path in `"Go to the work/01-tut-porting/libs/libhogweed/ directory and follow the [...]"`  to `"work/01-tut-porting/app-libhogweed/.unikraft/libs/libhogweed"`
> 
> 2. Change `newlib` reference  to `musl` 
> 

Done, thank you.

> Also, there seems to be an additional "Support files" section rendered

I'm looking into that now.

File: content/en/community/hackathons/usoc23/app-porting/index.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+### 12. Register Glue Code
+
+{{< readfile file="/community/hackathons/sessions/contributing-to-unikraft/content/Practical-Work_12.-Register-Glue-Code.md" markdown="true" >}}
+
+### 13. Final Verification
+
+{{< readfile file="/community/hackathons/sessions/contributing-to-unikraft/content/Practical-Work_13.-Final-Verification.md" markdown="true" >}}
+
+### 14. Give Us Feedback
```
Comment (eduardvintila) [inline content/en/community/hackathons/usoc23/app-porting/index.md:99]: This is a feedback form which is ancient - we should remove that.

File: content/en/community/hackathons/usoc23/app-porting/index.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Summary
+
+{{< readfile file="/community/hackathons/sessions/contributing-to-unikraft/content/Summary.md" markdown="true" >}}
+
+## Practical Work
+
+{{< readfile file="/community/hackathons/sessions/contributing-to-unikraft/content/Practical-Work.md" markdown="true" >}}
+
+### Support Files
```
Comment (eduardvintila) [inline content/en/community/hackathons/usoc23/app-porting/index.md:39]: This is the extra `Support Files` section I was talking about, there is an almost identical one above it

**Review by eduardvintila — COMMENTED** [review]
@StefanJum, the session looks great 🚀 - just a few more comments from my side.

Also, when building the `libhogweed` app I have problems when running `make` on multiple threads. Seems like a synchronization problem from the library repo's Makefile.uk:

```
  CONFIG  libhogweed: config.h
  CONFIG  libhogweed: make_auxiliar
make[3]: *** No rule to make target 'ecc-secp192r1.h'.  Stop.
```

A small note on this would be great. We can run `make` on a single thread just for the initial config part, afterwards when files are starting to be compiled we can run `make` again with multiple threads, to keep build times short.

**Comment by StefanJum** [comment]
Files @eduardvintila, thank you

**Review by eduardvintila — APPROVED** [review]
Thanks, @StefanJum - nice job 🚀 

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**